src/System.Windows.Forms/src/System/Windows/Forms/ICommandProvider.cs

```diff
@@ -0,0 +1,155 @@
+        /// <param name="newCommand">The new value of the <see cref="Command"/>
+        /// which should be assigned to the property.</param>
+        /// <param name="commandBackingField">The backing field for
+        /// the <see cref="Command"/> property.</param>
+        protected static void CommandSetter(
+            ICommandProvider commandComponent,
+            ICommand? newCommand,
+            ref ICommand? commandBackingField)
+            => commandComponent.CommandSetter(newCommand, ref commandBackingField);
+
+        /// <summary>
+        /// Method which should be called by the class implementing this
+        /// interface, when the assigned <see cref="Command"/> should be executed.
+        /// </summary>
+        /// <remarks>
+        /// As an example, a <see cref="Button"/> should call this method inside the method which
+        /// also raises the <see cref="Control.Click"/> Event of that Button, which
+        /// would be the <see cref="Button.OnClick(EventArgs)"/> OnClick method.
+        /// </remarks>
+        /// <param name="commandComponent"></param>
+        protected static void RequestCommandExecute(ICommandProvider commandComponent)
+        {
+            CancelEventArgs e = new();
+            commandComponent.RaiseCommandExecuting(e);
+
+            if (!e.Cancel && (commandComponent.Command?.CanExecute(commandComponent.CommandParameter) ?? false))
+            {
+                commandComponent.Command?.Execute(commandComponent.CommandParameter);
+            }
+        }
+
+        private void CommandSetter(
+            ICommand? newCommand,
+            ref ICommand? commandBackingField)
+        {
+            if (!Equals(commandBackingField, newCommand))
+            {
+                if (commandBackingField is not null)
+                {
+                    commandBackingField.CanExecuteChanged -= CommandCanExecuteChangedProc;
+
+                    // We need to restore Enabled, should we go from a defined Command to an undefined Command.
+                    if (newCommand is null)
+                    {
+                        if (PreviousEnabledStatus.HasValue)
+                        {
+                            Enabled = PreviousEnabledStatus.Value;
+                            PreviousEnabledStatus = null;
+                        }
+                    }
+                }
+
+                commandBackingField = newCommand;
+                RaiseCommandChanged(EventArgs.Empty);
+
+                if (commandBackingField is null)
+                {
+                    return;
+                }
+
+                commandBackingField.CanExecuteChanged += CommandCanExecuteChangedProc;
+                PreviousEnabledStatus ??= Enabled;
+                Enabled = commandBackingField.CanExecute(CommandParameter);
+            }
+        }
+
+        private void CommandCanExecuteChangedProc([AllowNull] object sender, EventArgs e)
```

Comment:
```suggestion
        private void Command_CanExecuteChanged(object? sender, EventArgs e)
```
ditto other places

Reply:
As dicussed you may wnat to verify this is all we need:
```suggestion
        private void Command_CanExecuteChanged(object? sender, EventArgs e)
        {
           Command?.RaiseCommandCanExecuteChanged();
           Enabled = Command?.CanExecute(CommandParameter) ?? false;
        }
```

Reply:
It's gone.